tests/rules/data/doctrine-key-value-style.php

```diff
@@ -119,4 +119,25 @@ public function errorInParamArrayUnion(Connection $conn, array $params)
     {
         $conn->assembleOneArray('ada', $params);
     }
+
+    /**
+     * @param int|float $value
+     */
+    public function errorWithUnacceptableUnionValue(Connection $conn, $value)
+    {
+        $conn->assembleOneArray('typemix', ['c_int' => $value]);
+    }
+
+    /**
+     * @param array-key $value A benevolent union type (int|string)
+     */
+    public function errorWithBenevolentUnionValue(Connection $conn, $value)
+    {
+        $conn->assembleOneArray('typemix', ['c_int' => $value]);
+    }
```

Comment:
I am not sure about this expectation. I would expect a error for a non-benevolent union, but I think we should be ok with a benevolent union, as long as it contains the type we expect

Reply:
I'm happy to change the behavior for benevolent unions as well. Would you be okay with me submitting it as a separate PR?

Reply:
fine